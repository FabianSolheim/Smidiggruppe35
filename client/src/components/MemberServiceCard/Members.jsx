import { Business } from "@mui/icons-material";
import { HandHoldingHeartIcon } from "../assets/handHoldingHeartIcon";

export const members = [
  {
    title: "Business",
    content:
      "For businesses who want \n" +
      "to use their resources \n" +
      "for greater social good",
    icon: <Business sx={{ fontSize: 50 }} />,
    linkTo: "/corporate",
  },
  {
    title: "Non-profits",
    content:
      "For nonprofits who are seeking \n" +
      "to carry out their frontline work \n",
    icon: <HandHoldingHeartIcon sx={{ fontSize: 50 }} />,
    linkTo: "/nonprofit",
  },
];
