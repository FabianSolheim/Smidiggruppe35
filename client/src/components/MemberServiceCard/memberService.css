

.service-container {
    display: flex;
    justify-content: center;

}

.service-header {
    text-align: center;
    font-size: 2em;
    color: #034F7A;

}

.service-content {
    text-align: center;
    font-size: 36px;
    color: #212121;
    padding: 1em 2em 2em 2em;
}

.service-icon {
    padding-top: 20px;
    color: #034F7A;
}

.service-card {
    padding: 1rem;
    margin: 100px;
    max-width: 500px;
    flex-direction: row;
    height: 400px;
    box-shadow: #034F7A;

}

.service-button {
    padding-top: 50px;

}



