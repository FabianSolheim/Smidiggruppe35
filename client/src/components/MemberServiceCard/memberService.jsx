import {Business, Soap} from "@mui/icons-material";
import {Box, Button, Card, CardActions, Grid, SvgIcon, Typography} from "@mui/material";
import "./memberService.css";
import {HandHoldingHeartIcon} from "../../assets/handHoldingHeartIcon";
import CustomButton from "../CustomButton/CustomButton";


export function MemberService() {
    return (
        <Box textAlign="center" className="service-container">
            <Grid container flexDirection="row">
            <Card className="service-card">
                <Business fontSize="large" className="service-icon"/>
                <h4 className="service-header">
                    Business
                </h4>
                <Typography className="service-content">
                    For businesses who are profitable <br/>
                    and want to use their resources <br/>
                    for greater social good
                </Typography>
                <Box textAlign="center" className="service-button">
                    <CustomButton to="/" variant="contained">Learn more</CustomButton>
                </Box>
            </Card>
            <Card className="service-card">
                <HandHoldingHeartIcon fontSize="large" className="service-icon" />
                <h4 className="service-header">
                    Non-profits
                </h4>
                <Typography className="service-content">
                    For nonprofits who are seeking <br/>
                    to carry out their frontline work <br/>
                    <br/>
                </Typography>
                    <Box textAlign="center" className="service-button">
                        <CustomButton to="/" variant="contained">Learn more</CustomButton>
                    </Box>
            </Card>
            </Grid>
        </Box>
    )
}