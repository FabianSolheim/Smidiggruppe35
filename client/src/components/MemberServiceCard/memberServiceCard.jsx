import { Box, Grid, Typography } from "@mui/material";
import "./memberService.css";
import "./Members";
import CustomButton from "../CustomButton/CustomButton";
import {Link} from "react-router-dom";

export function MemberServiceCard({ title, content, icon, linkTo }) {
  return (
    <Link to={linkTo}>
    <Box className="service-card" py={5} px={8}>
      <Grid
        container
        direction="column"
        alignItems="center"
        justifyContent="space-around"
      >
        <Grid item>
          <Box className="service-icon">
            <Typography align="center" color="primary" fontSize={50}>
              {icon}
            </Typography>
          </Box>
          <Typography
            align="center"
            gutterBottom
            color="textPrimary"
            variant="h3"
            className="service-header"
          >
            {title}
          </Typography>
          <Typography
            align="left"
            paragraph
            fontSize={16}
            className="service-content"
          >
            {content}{" "}
          </Typography>
        </Grid>
        <Box className="member-service-card__button">
          <CustomButton p="12px 22px" to={linkTo} variant="outlined">
            Learn more
          </CustomButton>
        </Box>
      </Grid>
    </Box>
    </Link>
  );
}
