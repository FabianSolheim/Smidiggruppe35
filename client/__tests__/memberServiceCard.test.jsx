import React from "react";
import ReactDOM from "react-dom";
import { MemberServiceCard } from "../src/components/MemberServiceCard/memberServiceCard";
import { MemoryRouter } from "react-router-dom";

describe("Member service Cards test", () => {
  it("shows member service card", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <MemberServiceCard
          content={"test"}
          icon={"test"}
          linkTo={"/test"}
          title={"test"}
        />
      </MemoryRouter>,
      domElement
    );
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
