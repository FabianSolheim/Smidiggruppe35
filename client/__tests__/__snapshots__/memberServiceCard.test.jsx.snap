// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member service Cards test shows member service card 1`] = `"<a href=\\"/test\\"><div class=\\"service-card MuiBox-root css-19wy109\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-rnlwrt-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><div class=\\"service-icon MuiBox-root css-0\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uretal-MuiTypography-root\\">test</p></div><h4 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter MuiTypography-gutterBottom service-header css-wwozti-MuiTypography-root\\">test</h4><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiTypography-paragraph service-content css-1i5fhu7-MuiTypography-root\\">test </p></div><div class=\\"member-service-card__button MuiBox-root css-0\\"><a class=\\" css-vcbgxq-MuiButton-root\\" component=\\"button\\" focusvisibleclassname=\\"\\" style=\\"padding: 12px 22px; text-decoration: none; font-weight: 500; border-radius: 0; text-transform: none;\\" classes=\\"[object Object]\\" href=\\"/test\\">Learn more</a></div></div></div></a>"`;
